### PR TITLE
Add miniforge_console_shortcut on windows

### LINKED ISSUE #62: Windows Installer does not provide a prompt, doesn't work in cmd
Hello!

Unless I am missing something, it doesn't seem that the Windows miniforge installer provides a command prompt that initializes in the base environment. Also, usage by path in a cmd.exe terminal fails to do anything real too. Here is a simple update example that cannot be run.  (Note that miniconda fails in the same way with usage by path).

```cmd
C:\Users\Anthony.Scopatz>miniforge\\Scripts\\conda.exe update --all                                                     
Collecting package metadata (current_repodata.json): failed   
                                                                                                                                                                                  
CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://conda.anaconda.org/conda-forge/win-64/current_repodata.json>                                                                                                                        
Elapsed: -                                                                                                  
                                                                                                                                    
An HTTP error occurred when trying to retrieve this URL.                                                                
HTTP errors are often intermittent, and a simple retry will get you on your way.                                        'https://conda.anaconda.org/conda-forge/win-64'                                                                                                                                                                                                                                                                                                                                                                                                                                                   
```

This error message is fully reproducible and does not have anything to do with my internet connection.  A `conda update --all` in a full miniconda shell works fine.

Perhaps some clarification could be given in the README on how miniforge on Windows is intended to be used.

### PR DESCRIPTION
**Pull request by ericpre**
Fix #62.

- Add `miniforge_console_shortcut` on windows to add start menu shortcut
- Upload installer as github artefact when workflow doesn't run on tag.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
I'm confused about the necessity of `menuinst` is it still required to get the shortcuts to appear on windows? or has that been built into conda

**Comment by ericpre** [comment]
`menuinst` is a dependency of conda. Also, there a few recipe using it on conda-forge, such as for example the [ovito](https://github.com/conda-forge/ovito-feedstock/blob/master/recipe/meta.yaml) recipe and the menuinst dependency is not listed explicitly and I don't see why it would be necessary.
Also, I have added an option to upload the installer as github artefacts, so we can [download the two windows installer](https://github.com/conda-forge/miniforge/suites/1745319544/artifacts/33151750) and test them: it works as expected.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -171,6 +191,12 @@ jobs:
         ls -alh build
       shell: bash
 
+    - name: Upload miniforge to Github artefact
+      if: ${{ matrix.UPLOAD_ARTEFACT && !startsWith(github.ref, 'refs/tags/') }}
+      uses: actions/upload-artifact@v2
+      with:
+        path: build/${{ matrix.MINIFORGE_NAME }}-${{ matrix.OS_NAME }}-${{ matrix.ARCH }}*
```
Comment (isuruf) [inline .github/workflows/ci.yml:198]:
```suggestion
        path: build/${{ matrix.MINIFORGE_NAME }}-${{ matrix.OS_NAME }}-${{ matrix.ARCH }}*
        name: ${{ matrix.MINIFORGE_NAME }}-${{ matrix.OS_NAME }}-${{ matrix.ARCH }}
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -13,125 +13,145 @@ jobs:
             ARCH: x86_64
             MINIFORGE_NAME: "Miniforge3"
             OS_NAME: "Windows"
+            UPLOAD_ARTEFACT: true
```
Comment (isuruf) [inline .github/workflows/ci.yml:16]: Let's remove these and upload artifacts always

Comment (ericpre) [reply .github/workflows/ci.yml:16]: Yes, I was annoyed with getting one large artefact but setting the name of the artefact as you suggested above fix it!

**Comment by isuruf** [comment]
Thanks

**Pull request merged**